arelle/ModelObject.py

```diff
@@ -255,8 +255,9 @@ def parentQname(self) -> QName | None:
 
 
     @property
-    def id(self) -> str | None:
-        return self.get("id")
+    def id(self) -> str:
+        _id = self.get("id")
+        return cast(str, _id)
```

Comment:
Sorry, my [previous comment](https://github.com/Arelle/Arelle/pull/328#discussion_r967912272) wasn't clear. `ModelContext` has to have an `id`, but there are other `ModelObject` sub classes which don't share that requirement.

Reply:
Looks like we didn't have to do an assert in `ESEF/__init__.py` when changing the property. At least not locally, let's see what the CI/CD says.